### DISCUSSION TITLE
add multiple experiments to one run file

### DISCUSSION THREAD

**Discussion by SRJongman** [jspsych/jsPsych#973] [Q&A] [answered]
Hi,

I have programmed several experiments and would like to have one file that calls and runs each experiment individually, and saves the data from each experiment separately. I have sort of managed the first by creating .js files for each experiment and pushing the different steps of each experiment to the timeline in the 'run' file but this way I can only save one file at the end, a simple version of my code is pasted below. Is there a better way of running and saving each experiment consecutively? 

Thanks very much for the course, it has really made programming these experiments easier! 

Suzanne

```html
<!DOCTYPE html>
<html>
  <head>
    <!-- load library, css for formatting, and the necessary plugins -->
    <title>Experiment</title> 
    <script src="jspsych-6/jspsych.js"></script>
    <script src="jspsych-6/plugins/jspsych-html-keyboard-response.js"></script>
    <script src="StroopTask.js"></script>
    <script src="NbackTask.js"></script>
    <link rel="stylesheet" href="jspsych-6/css/jspsych.css">
  </head>
  <body></body>
  <script>

var exp_timeline = [];

exp_timeline.push(stroop_instructions);
exp_timeline.push(stroop_procedure);
exp_timeline.push(nback_instructions);
exp_timeline.push(nback_procedure);

    jsPsych.init({
      timeline: exp_timeline,
      on_finish: function(){
        jsPsych.data.get().localSave('csv','Data.csv');
      }
    })
  </script>
</html>
```

**Comment by jodeleeuw** [comment, marked as answer]
Hi Suzanne,

One option for saving the data separately would be to stick a `call-function` trial at the end of each procedure that saves the data for that procedure. 

You could do this by using `jsPsych.data.get().filter(...)` to grab only the data associated with the procedure. The simplest filter to set up would probably involve adding a key-value pair to the `data` field for each trial in a procedure that identifies which procedure it is part of. Luckily, if you have all of the trials for a procedure on a timeline, as it looks like you do based on your code, you can add the `data` property to the timeline itself and it will propagate to all trials on the timeline.

```js
var stroop_procedure = {
	timeline: [...],
	data: {procedure: 'stroop'}
}
```

The `call-function` trial would look something like this.

```js
var save_stroop_data = {
	type: 'call-function',
	func: function(){
		jsPsych.data.get().filter({procedure: 'stroop'}).localSave('csv', 'stoop-data.csv')
	}
}
```
